### PR TITLE
Fix mapbuilder in editor

### PR DESCRIPTION
**Pull request by zeroZshadow**
#### Description
Refactored some bits of the map builder to work in both the editor and play mode again.

#### Tests
Testing in a stand alone build atm

### CODE REVIEW

**Comment by zeroZshadow** [comment]
Tested in standalone client, no issues

File: workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs [hunk 1]
```diff
@@ -81,45 +84,44 @@ public IEnumerator CleanAndBuild(int worldLayers = 4, string seed = "SpatialOS G
...
-
-            mapTemplate.defaultTileCollection.LoadAndOptimizeTiles();
-            SceneManager.SetActiveScene(activeScene);
+                foreach (var tileCollection in mapTemplate.tileCollections)
+                {
+                    tileCollection.LoadAndOptimizeTiles();
+                }
 
-            yield return null;
+                mapTemplate.defaultTileCollection.LoadAndOptimizeTiles();
```
Comment (jessicafalk) [inline workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs]: why did that `yield return null;` go away

File: workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs [hunk 2]
```diff
@@ -442,5 +445,31 @@ public static Vector3 GetWorldDimensions(int layers)
...
         }
+
+        private class SceneScope : IDisposable
+        {
+            private readonly Scene activeScene;
+
+            public SceneScope(Scene newActive)
+            {
+#if UNITY_EDITOR
+                if (Application.isPlaying)
```
Comment (jessicafalk) [inline workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs]: why do we need that? is `Application.isPlaying` only available in the editor?

File: workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs (hunk 1 above)
Comment (zeroZshadow) [reply workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs]: This was an unneeded frame wait, removing it allowed me to simplify the code significantly

File: workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs (hunk 2 above)
Comment (zeroZshadow) [reply workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs]: No, but there is no reason to do this test in standalone builds

File: workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs (hunk 1 above)
Comment (jessicafalk) [reply workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs]: is it also unneeded on mobile :D cause I don't want it to look like it's hanging

File: workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs (hunk 2 above)
Comment (jessicafalk) [reply workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs]: but that code just doesn't look nice :(

File: workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs [hunk 3]
```diff
@@ -281,36 +283,37 @@ private IEnumerator PlaceTiles()
 
             // Tiles are built in a spiral manner from the centre outward to ensure increasing the # of tile layers doesn't
             // alter the existing tile types.
-            SceneManager.SetActiveScene(tempScene);
-            for (var i = 0; i < tileCount; i++)
+            using (new SceneScope(tempScene))
```
Comment (jamiebrynes7) [inline workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs]: This does nothing

**Pull request merged**